File: Website/src/types/article.ts
import { CommentType } from "./types";

export type ArticleType = {
	id: number
	title: string
	content: string
	category_name: string
	image: null | string
	created_by: number
	created_by_login: string
	modified_by: number
	is_published: boolean
	created_at: string
	updated_at: string
	rating: number
	count_comments: number

	vote: null | string
	comments?: Array<CommentType>
}

File: android/BettingHUB/app/src/main/java/com/bettinghub/forecasts/models/Article.kt
package com.bettinghub.forecasts.models

import com.google.gson.annotations.SerializedName

class Article(
    val id: Int,
    val image: String,
    @SerializedName("category_name") val categoryName: String,
    @SerializedName("created_at") val createdAt: String,
    val title: String,
    val content: String,
    val commentCount: Int,
    var rating: Int
)
